src/messages/InvMessage.cpp

```diff
@@ -0,0 +1,144 @@
+            const CInv &inv = vInv[nInv];
+
+            boost::this_thread::interruption_point();
+            pfrom->AddInventoryKnown(inv);
+
+            bool fAlreadyHave = AlreadyHave(inv);
+            logDebug(Log::Net) << "got inv:" << inv << (fAlreadyHave ? "have" : "new") << "peer:" << pfrom->id;
+
+            if (inv.type == MSG_BLOCK) {
+                UpdateBlockAvailability(pfrom->GetId(), inv.hash);
+                if (!fAlreadyHave && !fImporting && !fReindexing && !mapBlocksInFlight.count(inv.hash)) {
+                    // First request the headers preceding the announced block. In the normal fully-synced
+                    // case where a new block is announced that succeeds the current tip (no reorganization),
+                    // there are no such headers.
+                    // Secondly, and only when we are close to being synced, we request the announced block directly,
+                    // to avoid an extra round-trip. Note that we must *first* ask for the headers, so by the
+                    // time the block arrives, the header chain leading up to it is already validated. Not
+                    // doing this will result in the received block being rejected as an orphan in case it is
+                    // not a direct successor.
+                    pfrom->PushMessage(NetMsgType::GETHEADERS, chainActive.GetLocator(pindexBestHeader), inv.hash);
+                    const CChainParams chainparams = Params();
+                    CNodeState *nodestate = State(pfrom->GetId());
+                    if (CanDirectFetch(chainparams.GetConsensus()) &&
+                        nodestate->nBlocksInFlight < MAX_BLOCKS_IN_TRANSIT_PER_PEER) {
+                        // BUIP010 Xtreme Thinblocks: begin section
+                        CInv inv2(inv);
+                        CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
+                        if (xthinEnabled && IsChainNearlySyncd()) {
+                            if (HaveThinblockNodes() && CheckThinblockTimer(inv.hash)) {
+                                // Must download a block from a ThinBlock peer
+                                if (pfrom->mapThinBlocksInFlight.size() < 1 &&
+                                    pfrom->ThinBlockCapable()) { // We can only send one thinblock per peer at a time
+                                    pfrom->mapThinBlocksInFlight[inv2.hash] = GetTime();
+                                    inv2.type = MSG_XTHINBLOCK;
+                                    CBloomFilter filterMemPool = createSeededBloomFilter(
+                                            CTxOrphanCache::instance()->fetchTransactionIds());
+                                    ss << inv2;
+                                    ss << filterMemPool;
+                                    pfrom->PushMessage(NetMsgType::GET_XTHIN, ss);
+                                    MarkBlockAsInFlight(pfrom->GetId(), inv.hash, chainparams.GetConsensus());
```

Comment:
You forgot to declare the MarkBlockAsInFlight in the main.h

Which itself requires you move the namespace {} markers it is caught in right now so it can be used outside of the compile-unit.